base_agent/dialogue_objects/interpreter.py

```diff
@@ -85,17 +113,42 @@ def step(self) -> Tuple[Optional[str], Any]:
                 actions.append(self.action_dict["action"])
             elif "action_sequence" in self.action_dict:
                 actions = self.action_dict["action_sequence"]
-                actions.reverse()
 
             if len(actions) == 0:
                 # The action dict is in an unexpected state
                 raise ErrorWithResponse(
                     "I thought you wanted me to do something, but now I don't know what"
                 )
+            tasks_to_push = []
             for action_def in actions:
                 action_type = action_def["action_type"]
-                response = self.action_handlers[action_type](self.speaker, action_def)
-            return response
+                try:
+                    r = self.action_handlers[action_type](self.speaker, action_def)
+                    if len(r) == 3:
+                        tasks, response, dialogue_data = r
+                    else:
+                        # FIXME don't use this branch, uniformize the signatures
+                        tasks = []
+                        response, dialogue_data = r
+                except ErrorWithResponse as err:
+                    return err.chat, None
+                tasks_to_push.extend(tasks)
+            if len(tasks_to_push) == 1:
+                TaskMem = TaskNode(self.agent.memory, tasks_to_push[0].memid)
+            elif len(tasks_to_push) > 1:
+                task_data = {"new_tasks_fn": tasks_to_push}
+                c = ControlBlock(self.agent, task_data)
+                TaskMem = TaskNode(self.agent.memory, c.memid)
+            else:
+                TaskMem = None
+            if TaskMem:
+                TaskMem.get_update_status({"prio": 1})
```

Comment:
we should document the possible values of `prio` , what they represent and perhaps call it `priority` for better variable naming